pkgs/racket-doc/scribblings/reference/match.scrbl

```diff
@@ -758,14 +758,26 @@ This binding is provided @racket[for-syntax].
 @defproc[(legacy-match-expander? [v any/c]) boolean?]]]{
 Predicates for values which implement the appropriate match expander
 properties.
+These bindings are provided @racket[for-syntax].
 }
 
 @defproc[(syntax-local-match-introduce [stx syntax?]) syntax?]{
 For backward compatibility only; equivalent to @racket[syntax-local-introduce].
 
+This binding is provided @racket[for-syntax].
+
 @history[#:changed "6.90.0.29" @elem{Made equivalent to @racket[syntax-local-introduce].}]}
 
 
+@defproc[(pattern-bound-identifiers #:legacy? [legacy? any/c #f] [pat syntax?]) (listof identifier?)]{
```

Comment:
Is there a benefit to allowing non-boolean values for `#:legacy?`? It doesn't seem worth making the API harder to understand.

Reply:
What do you think of renaming this to `match-pattern-bound-identifiers`? Since we've also got `syntax-case` and `syntax-parse` patterns.

Reply:
Allowing generalized booleans seems conventional, which is why I'm doing it.

Reply:
I like that renaming; I'll try to update soon.

Reply:
For the sake of being consistent with the rest of Racket, I think it should allow non-boolean.

See various string functions that accept keyword arguments like `#:all?`, `#:trim?`, etc. as examples

```
(string-replace str from to [#:all? all?]) → string?
str : string?
from : (or/c string? regexp?)
to : string?
all? : any/c = #t
```

Reply:
Alright, `any/c` on the basis of consistency seems reasonable to me. It still strikes me as weird though to allow non-booleans in places where there's no expectation that any useful data could be passed through. Probably too late to rock that boat for core Racket APIs.